### DISCUSSION TITLE
[Section 4.21 - Workspaces] - Instâncias em diferentes Availability Zones

### DISCUSSION THREAD

**Discussion by chgasparoto** [chgasparoto/curso-aws-com-terraform#40] [Q&A]
Cleber que fantastica sua didatica e abordagem do tema, comprei o seu curso na Udemy e gostaria de tirar uma duvida com vc sobre workspace e resolvi perguntar aqui mais vou fazer a pergunta no curso tb pra ajudar a galera, tudo bem?

Se eu usar workspaces com 3 ambientes dev, hom e prod e eu tiver 2 regioes de disponibilidade e meu main estiver assim:

```hcl
count              = lookup(var.instance, local.env)["number"]  // ----> exemplo da aula ok?
availability_zone  = ? 
```

se eu tiver que subir 2 ou três maquinas no ambiente como faço pra subir uma em cada availability_zone ?

#####################################################################################
Eu usava assim antes:

main.tf

```hcl
count                  = "${lookup(var.instance_count,terraform.workspace)}
availability_zone      = "${element(split(",", var.availability_zones), count.index)}"
```

vars.tf

```hcl
variable "instance_count" {
  type = "map"
  default = {
    prod = 3
    hom  = 3
    dev  = 1
  }
}

variable "availability_zones" {
    default = "us-east-2a,us-east-2c"
}
```

Por favor consegue tirar essa dúvida, fui claro na pergunta?
Mais uma vez obrigado seu curso e videos aqui no Youtube estão me ajudando muito..

_Originally posted by **Angelo Moreira** on Youtube (https://www.youtube.com/watch?v=e7yk5aY9evo&ab_channel=CleberGasparoto)_

**Comment by chgasparoto** [comment]
Angelo,

Muito obrigado pelas palavras de apoio!

Eu sugiro você a usar o `foreach` ao invés do `count`, pois acho mais fácil de manipular objetos mais complexos.

Dá uma olhada nessa aula que você vai entender o que eu quero dizer.

**Aula**: https://www.udemy.com/course/aws-com-terraform/learn/lecture/25939516#overview
**Código Fonte**: https://github.com/chgasparoto/curso-aws-com-terraform/blob/master/05-autoscaling-app/vpc.tf#L11

**Reply by Ang3l1nux** [reply]
Valeu Cleber obrigado e muito sucesso com seu trabalho, vc está ajudando diversas pessoas contribuindo com a passagem de conhecimento. Sucesso sempre !!!!
